crates/pathfinder/src/consensus/inner/consensus_task.rs

```diff
@@ -99,8 +99,18 @@ pub fn spawn(
                 highest_finalized,
             )?;
 
-        // Get the current height
-        let mut current_height = consensus.current_height().unwrap_or_default();
+        // Compute the next height to work on using all available information:
+        // - max_active_height: highest incomplete/active height being tracked
+        // - last_decided_height: highest decided height (even if not actively tracked)
+        // - highest_finalized + 1: next height after what's been committed to DB
+        let mut next_height = [
+            consensus.max_active_height().unwrap_or(0),
+            consensus.last_decided_height().unwrap_or(0),
```

Comment:
If `last_decided_height()` is higher than `highest_finalized`, then `max_active_height()` == `last_decided_height()` (because of https://github.com/eqlabs/pathfinder/blob/362805eb5ea823abcf9584f118b87e5b14ef7f88/crates/consensus/src/wal.rs#L300 ) - IOW `last_decided_height()` isn't needed here.

Reply:
> This is not always true.
> 
> Here's an example scenario simulating the edge case pointed out by @CHr15F0x and outlined in my original [#3096 (comment)](https://github.com/eqlabs/pathfinder/pull/3096#issuecomment-3526064038):
> 
>     1. Height `H` decided (`Decision` in WAL)
> 
>     2. Height `H+1` incomplete (proposals/votes, no `Decision` yet)
> 
>     3. `highest_finalized` = `H-1`
> 
> 
> On recovery:
> 
>     * `H` has `Decision` → skipped (not added to `internal`)
> 
No - it is added to `internal`, because `highest_finalized` = `H-1` < `H`, so the test referenced above fails and `is_finalized` is set to false. I agree the behavior is surprising... :-)